snap-app/src/wordsearch/wordsearch.js

```diff
@@ -70,54 +114,142 @@ export default class Wordsearch extends React.Component {
                     value={word}
                     readOnly={true}
                     onMouseEnter={() => {
-                        const highlightedPositions = {};
-                        for (const { x, y } of positions) {
-                            highlightedPositions[`${x}-${y}`] = true;
+                        if (!highlightingAllWordbank) {
+                            const highlightedPositions = {};
+                            const highlightedWords = {};
+                            for (const { x, y } of positions) {
+                                highlightedPositions[`${x}-${y}`] = { inWordbank };
+                            }
+                            this.setState({ highlightedIndex: index, highlightedPositions, highlightedWords });
                         }
-                        this.setState({ highlightedIndex: index, highlightedPositions });
                     }}
                 />)}
             </div>
         </div>;
     }
 
+    hashLocation() {
+        const { grid, wordbank, boggle, highlightingAllWordbank, highlightingDifferentColors } = this.state;
+        window.location.hash = Buffer.from(JSON.stringify({ grid, wordbank, boggle, highlightingAllWordbank, highlightingDifferentColors })).toString('base64');
+    }
+
+    hashStringToColor(str) {
+        var hash = 0;
+        for (let i = 0; i < str.length; i++) {
+            hash = str.charCodeAt(i) + ((hash << 5) - hash);
+        }
+        let colour = '#';
+        for (let i = 0; i < 3; i++) {
+            const value = (hash >> (i * 8)) & 0xFF;
+            colour += ('00' + value.toString(16)).substr(-2);
+        }
+        colour += 'aa';
+        return colour;
+    }
+
+    onHighlightWordbankChange() {
+        const { highlightingAllWordbank, highlightingDifferentColors, results } = this.state;
+        const highlightedWords = {};
+        const highlightedPositions = {};
+        if (highlightingAllWordbank) {
+            results.filter(result => result.inWordbank).forEach(
+                ({ positions, word }, index
+            ) => {
+                const color = highlightingDifferentColors ? this.hashStringToColor(word) : '#f9ce70';
+                highlightedWords[index] = { color };
+                positions.forEach(({ x, y }) => {
+                    const key = `${x}-${y}`;
+                    if (!highlightedPositions.hasOwnProperty(key)) {
+                        highlightedPositions[key] = { color };
+                    }
+                });
+            });
+        }
+        this.setState({ highlightedPositions, highlightedWords, highlightedIndex: undefined });
+        this.hashLocation();
+    }
+
     getGrid() {
         return this.state.grid.trim().toUpperCase().split('\n').map(word => word.replace(/\s/g, ''));
     }
 
+    getWordbank() {
+        return this.state.wordbank.trim().toUpperCase().split(/\n|\t/).map(word => word.replace(/[^A-Z0-9]/g, '')).filter(word => word.length > 1);
```

Comment:
Doesn't really matter, but I wouldn't expect this to filter out one-letter words
```suggestion
        return this.state.wordbank.trim().toUpperCase().split(/\n|\t/).map(word => word.replace(/[^A-Z0-9]/g, '')).filter(word => word.length >= 1);
```